lib/sles4sap/sdaf_library.pm

```diff
@@ -609,48 +621,97 @@ sub replace_tfvars_variables {
     file_content_replace($tfvars_file, %to_replace);
 }
 
-=head2 sdaf_deploy_workload_zone
+=head2 sdaf_execute_deployment
+
+    sdaf_execute_deployment(deployment_type=>$deployment_type [, timeout=>$timeout]);
 
-    sdaf_deploy_workload_zone();
+B<deployment_type>: Type of the deployment: workload_zone or sap_system
 
-Executes SDAF workload zone deployment. SDAF relies on OS env variables therefore those are passed as cmd args as well.
-Definitely keep I<--spn-secret> set as a reference to OS variable I<${ARM_CLIENT_SECRET}>, otherwise password will
-be shown in openQA output log in plaintext.
+B<timeout>: Execution timeout. Default: 1800s.
+
+Executes SDAF deployment according to the type specified.
+Croaks with unsupported deployment type, dies upon command failure.
 L<https://learn.microsoft.com/en-us/azure/sap/automation/deploy-workload-zone?tabs=linux#deploy-the-sap-workload-zone>
+L<https://learn.microsoft.com/en-us/azure/sap/automation/tutorial#deploy-the-sap-system-infrastructure>
 
 =cut
 
-sub sdaf_deploy_workload_zone {
```

Comment:
Generalizing function so it can be used for both workload zone and sap systems deployment.